monitor/routerecon/recon.go

```diff
@@ -0,0 +1,260 @@
+// Package routerecon defines functions to reconcile routescan cross-transactions verifying the data
+// matches expected on-chain values.
+package routerecon
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/omni-network/omni/e2e/app/eoa"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+func ReconForever(ctx context.Context, network netconf.Network, xprov xchain.Provider, ethCls map[uint64]ethclient.Client) {
+	if network.ID != netconf.Omega {
+		return
+	}
+
+	ticker := time.NewTicker(time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			crossTx, err := paginateLatestCrossTx(ctx)
+			if err != nil {
+				reconFailure.Inc()
+				log.Warn(ctx, "RouteRecon failed fetching latest cross transaction (will retry)", err)
+
+				continue
+			}
+
+			if err := reconOnce(ctx, network, xprov, ethCls, crossTx); err != nil {
+				reconFailure.Inc()
+				log.Warn(ctx, "RouteRecon failed (will retry)", err, "id", crossTx.ID)
+
+				continue
+			}
+
+			reconSuccess.Inc()
+			log.Info(ctx, "RouteRecon success", "id", crossTx.ID)
+		}
+	}
+}
+
+func reconOnce(
```

Comment:
@fabtreb see here for what is reconned

Reply:
I basically verify all the data returned by the routescan API:
```
 {
      "id": "ac147cda3a08dc9720051e05487ed6c7b090fb4f66dfff9cdd08b87f0a0f923a",
      "type": "omni",
      "status": "completed",
      "srcChainId": "164_4",
      "srcTimestamp": "2024-08-20T12:52:22.000Z",
      "srcTxHash": "0x8e8554851ab2187d694da551330450f9513168347083f52a702464ef442d69d4",
      "srcBlockNumber": 295862,
      "srcBlockHash": "0xcf83778f5b01b14d613b2ae289de80ad5da5fec358e301fca5a12bd2f47a30c2",
      "srcGasLimit": "1400000",
      "dstChainId": "84532",
      "dstTimestamp": "2024-08-20T12:52:30.000Z",
      "dstTxHash": "0xa1d39b62214b72618c07ca7ddde5c864c3a46ca1be8c6a86ddf91596df3f4ead",
      "dstBlockNumber": 14195031,
      "dstBlockHash": "0xa5b781dd90c6db2f06485444a1f6e3e70427269380164870e468fa4f1e28bc16",
      "dstGasLimit": "1400000",
      "from": "0xE30C640c5fcc87dCcff7522D66262d166948B803",
      "to": "0x918b8E25EAc626CFb1080D704Be5f5851D9B629d",
      "data": {
        "relayer": "0x37AD6f7267454cac494C177127aC017750c8A7DB",
        "gasUsed": "97442",
        "error": "0x",
        "offset": "1078",
        "shardId": "4",
        "confirmationLevel": "finalized",
        "gasLimit": "200000",
        "fees": "101761408306048"
      }
    }
```

Reply:
Thanks @corverroos. Is there a way to confirm all txs are on the explorer from day0? Can we do an overall count comparison?

Reply:
I went with the proposed solution: "No need to check 100% of the data, simple periodic sampling should be fine."

A full comparison would require much more work, and archive nodes of all chains.

Reply:
Ok sounds good for now